### PR TITLE
Applications Support

### PR DESCRIPTION
**Pull request by piyumaldk**
## Purpose
$Subject

### CODE REVIEW

File: src/controllers/applicationsContentController.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    ca: fs.readFileSync(certPath),  
+    rejectUnauthorized: true,
+});
+
+// ***** Load Applications *****
+
+const loadApplications = async (req, res) => {
+    const orgName = req.params.orgName;
+    let html, metaData, templateContent;
+    if (config.mode === constants.DEV_MODE) {
```
Comment (SachiniSiriwardene) [inline src/controllers/applicationsContentController.js:26]: No need of a dev mode for applications

File: src/controllers/applicationsContentController.js [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        metaData = await getMockApplications();
+        templateContent = {
+            applicationsMetadata: metaData,
+            baseUrl: constants.BASE_URL + config.port
+        }
+    }
+    else {
+        console.log('/' + orgName);
```
Comment (SachiniSiriwardene) [inline src/controllers/applicationsContentController.js:34]: remove unnecessary logs

File: src/controllers/applicationsContentController.js [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+const httpsAgent = new https.Agent({
+    ca: fs.readFileSync(certPath),  
+    rejectUnauthorized: true,
+});
+
+// ***** Load Applications *****
+
+const loadApplications = async (req, res) => {
+    const orgName = req.params.orgName;
+    let html, metaData, templateContent;
```
Comment (SachiniSiriwardene) [inline src/controllers/applicationsContentController.js:25]: lets use const where possible

File: src/controllers/applicationsContentController.js [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+        });
+        return response.data.list;
+    } catch (error) {
+        console.error('Error fetching throttling policies:', error.message);
+    }
+}
+
+// ***** Load Application *****
+
+const loadApplication = async (req, res) => {
```
Comment (SachiniSiriwardene) [inline src/controllers/applicationsContentController.js:114]: why do we need two methods loadApplication and loadApplicationForEdit?

File: src/pages/application/partials/throttling-reset-modal.hbs [hunk 5]
```diff
@@ -0,0 +1,68 @@
+<head>
+  <link rel='stylesheet' href='/technical-styles/throttling-reset-modal.css' />
+  <script src='/technical-scripts/application-dashboard.js' defer></script>
+</head>
+
+<body>
+  <section>
+
```
Comment (SachiniSiriwardene) [inline src/pages/application/partials/throttling-reset-modal.hbs:8]: remove unnecessary empty lines

File: src/routes/applicationsContentRoute.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+const router = express.Router();
+const applicationsController = require('../controllers/applicationsContentController');
+const registerPartials = require('../middlewares/registerPartials');
+
+router.get('/((?!favicon.ico)):orgName/applications', registerPartials, applicationsController.loadApplications);
+router.get('/((?!favicon.ico)):orgName/applications/create', registerPartials, applicationsController.loadThrottlingPolicies);
+router.get('/((?!favicon.ico)):orgName/applications/:applicationid', registerPartials, applicationsController.loadApplication);
+router.get('/((?!favicon.ico)):orgName/applications/:applicationid/edit', registerPartials, applicationsController.loadApplicationForEdit);
+
+router.post('/applications', applicationsController.saveApplication);
```
Comment (SachiniSiriwardene) [inline src/routes/applicationsContentRoute.js:11]: what is the purpose of these routes (without the orgName)

**Comment by piyumaldk** [comment]
Closed since https://github.com/wso2/api-developer-portal-core/pull/59 is merged

File: src/controllers/applicationsContentController.js (hunk 1 above)
Comment (piyumaldk) [reply src/controllers/applicationsContentController.js:26]: Decided to keep those for now

File: src/controllers/applicationsContentController.js (hunk 2 above)
Comment (piyumaldk) [reply src/controllers/applicationsContentController.js:34]: +1

File: src/controllers/applicationsContentController.js (hunk 3 above)
Comment (piyumaldk) [reply src/controllers/applicationsContentController.js:25]: Those variables are not constants and values get assigned to them later

File: src/controllers/applicationsContentController.js (hunk 4 above)
Comment (piyumaldk) [reply src/controllers/applicationsContentController.js:114]: loadApplicationForEdit is being used inside loadApplication for one flow. This is to improve readability

File: src/pages/application/partials/throttling-reset-modal.hbs (hunk 5 above)
Comment (piyumaldk) [reply src/pages/application/partials/throttling-reset-modal.hbs:8]: +1

File: src/routes/applicationsContentRoute.js (hunk 6 above)
Comment (piyumaldk) [reply src/routes/applicationsContentRoute.js:11]: These are not get methods. These routes are being used to get methods to backend from frontend.

**Pull request closed without merging**